### ISSUE TITLE
Discussion: Should props also include tabIndex and onKeyDown?

### ISSUE THREAD

**Issue by tansongyang** [kentcdodds/react-toggled#5]
First of all, thanks for making this. I really like the direction that this project and downshift are taking.

I tried my hand at making a little [switch component](https://codesandbox.io/s/mqj1lnm69j). You'll see that I call `getTogglerProps` on an `<input>`. I then hide it under the control that I want to be displayed. I do this because it lets me tab to the control and toggle with the spacebar.

I originally wanted to be able to do [something like this](https://codesandbox.io/s/xpr1nqpo04). This does not use a hidden `<input>`. It still works with clicking, but you can't tab to it.

**My question**: Do you think it is within the scope of this project to get tabbing for free?

I've glanced at [getTogglerProps](https://github.com/kentcdodds/react-toggled/blob/master/src/index.js). I think you could achieve this by adding the following props.

```JavaScript
{
  tabIndex: props.tabIndex == null ? 0 : props.tabIndex, // In case users wanted a specific value
  onKeyDown: callAll(props.onKeyDown, this.toggleIfCorrectKey), // I'm open to a better name
}
```

If I'm not mistaken, this should keep existing functionality for inputs, buttons, etc., while allowing raw `<span>`s and other elements to gain new functionality.

**Comment by kentcdodds** [comment]
Yeah, I think that's a good idea :+1: Let's do it!

Happy to accept a pull request for it :)

**Comment by svinkle** [comment]
@tansongyang After checking out [your demo](https://codesandbox.io/s/xpr1nqpo04) and seeing what you want to accomplish, applying `tabindex="0"` won't be enough. The control will still be missing its role, state, and accessible name to convey its purpose.

You'll need to pass the following as well:
- `role="switch"` to provide semantic meaning for the element ("this is a _switch_ control")
- `aria-checked="true"` to convey it's current state, toggling to `false` (_"on"_ or _"off"_)
- `aria-label="[some label]"` to convey the purpose of the control

You'll also need to make sure to support `Enter` and `Space` key events as these are the expected methods of interacting with a "switch" control.

If you want to save some time and code, swap the `span` for a `button` which gives you keyboard focusability plus `Enter` and `Space` key support for free.

**Comment by tansongyang** [comment]
@kentcdodds Sounds good. I'd be happy to make a PR.

@svinkle Thanks. That's why I'm submitting this as an enhancement request.

**Comment by tansongyang** [comment]
Can we safely not care about IE9 and some older FireFox versions? They report weird values for the spacebar keydown event.

**Comment by kentcdodds** [comment]
Yeah, I don't care about IE9 and old Firefox. IE10 and up is what I want to support :+1:

**Comment by tansongyang** [comment]
Closed by #6.
